rfcs/0185-redistribute-redistributable.md

```diff
@@ -0,0 +1,108 @@
+---
+feature: redistribute-redistributable
+start-date: 2024-12-15
+author: Ekleog
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: https://github.com/NixOS/nixpkgs/issues/83884
+---
+
+# Summary
+[summary]: #summary
+
+Make Hydra build and provide all redistributable software.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, Hydra builds only free software and unfree redistributable firmware.
+This means that unfree redistributable software needs to be rebuilt by all the users.
+For example, using MongoDB on a Raspberry Pi 4 (aarch64, which otherwise has access to hydra's cache) takes literally days and huge amounts of swap.
+
+Hydra could provide builds for unfree redistributable software, at minimal added costs.
+This would make life much better for users of such software.
+Especially when the software is still source-available even without being free software, like MongoDB.
+
+# Detailed design
+[design]: #detailed-design
+
+Hydra will build all packages with licenses for which `redistributable = true`.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+With this change, Hydra will start building, among others:
+- CUDA
+- DragonflyDB
+- MongoDB
+- Nomad
+- NVIDIA drivers
+- Outline
+- SurrealDB
+- TeamSpeak
+- Terraform
+- Unrar
+- Vagrant
+
+Hydra will keep not building, among others:
+- CompCert
+- DataBricks
+- Elasticsearch
+- GeoGebra
+- Widevine CDM
```

Comment:
It would be helpful to add an explanation to some of those.

For example, Elasticsearch is in fact distributable and it's not clear why the license is marked as non-distributable in Nixpkgs.

Perhaps this is indicative of some status that needs to have its own separate parameter, instead of being conflated with existing ones.

Reply:
Might be, I don't know why the choice was made this way, it could just be a mistake. I'm updating the RFC to clarify that this list is indicative only and not a spec :)